# Lab | Hypothesis Testing

**Objective**

Welcome to the Hypothesis Testing Lab, where we embark on an enlightening journey through the realm of statistical decision-making! In this laboratory, we delve into various scenarios, applying the powerful tools of hypothesis testing to scrutinize and interpret data.

From testing the mean of a single sample (One Sample T-Test), to investigating differences between independent groups (Two Sample T-Test), and exploring relationships within dependent samples (Paired Sample T-Test), our exploration knows no bounds. Furthermore, we'll venture into the realm of Analysis of Variance (ANOVA), unraveling the complexities of comparing means across multiple groups.

So, grab your statistical tools, prepare your hypotheses, and let's embark on this fascinating journey of exploration and discovery in the world of hypothesis testing!

**Challenge 1**

In this challenge, we will be working with pokemon data. The data can be found here:

- https://raw.githubusercontent.com/data-bootcamp-v4/data/main/pokemon.csv

In [1]:
#libraries
import pandas as pd
import scipy.stats as st
import numpy as np

In [79]:
df = pd.read_csv("https://raw.githubusercontent.com/data-bootcamp-v4/data/main/pokemon.csv")
df.head(50)

df[df["Legendary"]==False]

Name  Type 1  Type 2  HP  Attack  Defense  Sp. Atk  \
0               Bulbasaur   Grass  Poison  45      49       49       65   
1                 Ivysaur   Grass  Poison  60      62       63       80   
2                Venusaur   Grass  Poison  80      82       83      100   
3           Mega Venusaur   Grass  Poison  80     100      123      122   
4              Charmander    Fire     NaN  39      52       43       60   
..                    ...     ...     ...  ..     ...      ...      ...   
787  Gourgeist Super Size   Ghost   Grass  85     100      122       58   
788              Bergmite     Ice     NaN  55      69       85       32   
789               Avalugg     Ice     NaN  95     117      184       44   
790                Noibat  Flying  Dragon  40      30       35       45   
791               Noivern  Flying  Dragon  85      70       80       97   

     Sp. Def  Speed  Generation  Legendary  
0         65     45           1      False  
1         80     60           1      False  
2        100     80           1      False  
3        120     80           1      False  
4         50     65           1      False  
..       ...    ...         ...        ...  
787       75     54           6      False  
788       35     28           6      False  
789       46     28           6      False  
790       40     55           6      False  
791       80    123           6      False  

[735 rows x 11 columns]

- We posit that Pokemons of type Dragon have, on average, more HP stats than Grass. Choose the propper test and, with 5% significance, comment your findings.

In [49]:
#code here

#Set hypothesis

#H0: Dragon average HP > Grass average HP
#H1: Dragon average HP < Grass average HP

#with alpha = 0.05

dragon_hp = df[(df["Type 1"]=="Dragon")|(df["Type 2"]=="Dragon")]["HP"]
grass_hp = df[(df["Type 1"]=="Grass")|(df["Type 2"]=="Grass")]["HP"]

dragon_hp
grass_hp


0      45
1      60
2      80
3      80
48     45
       ..
783    59
784    65
785    55
786    75
787    85
Name: HP, Length: 95, dtype: int64

In [50]:
alpha = 0.05
st.ttest_ind(dragon_hp,grass_hp, equal_var=False)
# accepet the null hypothesis

TtestResult(statistic=4.097528915272702, pvalue=0.00010181538122353852, df=77.58086781513519)

- We posit that Legendary Pokemons have different stats (HP, Attack, Defense, Sp.Atk, Sp.Def, Speed) when comparing with Non-Legendary. Choose the propper test and, with 5% significance, comment your findings.


In [80]:
#code here
#Set hypothesis

#H0: Legendary HP, Attack, Defense, Sp.Def, Speed
# Grass average HP
#H1: Legendary average HP < Grass average HP

#with alpha = 0.05

# df[['Name', 'Age']]
legendary = df[(df["Legendary"]==True)][['HP', 'Attack','Defense','Sp. Atk','Sp. Def','Speed']]
# non_legendary = df[(df["Legendary"]==False)]
non_legendary = df[(df["Legendary"]==False)][['HP', 'Attack','Defense','Sp. Atk','Sp. Def','Speed']]




In [84]:
display(non_legendary)
display(legendary)

HP  Attack  Defense  Sp. Atk  Sp. Def  Speed
0    45      49       49       65       65     45
1    60      62       63       80       80     60
2    80      82       83      100      100     80
3    80     100      123      122      120     80
4    39      52       43       60       50     65
..   ..     ...      ...      ...      ...    ...
787  85     100      122       58       75     54
788  55      69       85       32       35     28
789  95     117      184       44       46     28
790  40      30       35       45       40     55
791  85      70       80       97       80    123

[735 rows x 6 columns]

HP  Attack  Defense  Sp. Atk  Sp. Def  Speed
156   90      85      100       95      125     85
157   90      90       85      125       90    100
158   90     100       90      125       85     90
162  106     110       90      154       90    130
163  106     190      100      154      100    130
..   ...     ...      ...      ...      ...    ...
795   50     100      150      100      150     50
796   50     160      110      160      110    110
797   80     110       60      150      130     70
798   80     160       60      170      130     80
799   80     110      120      130       90     70

[65 rows x 6 columns]

In [83]:
alpha = 0.05
st.ttest_ind(legendary,non_legendary, equal_var=True)
# reject the null hypothesis

TtestResult(statistic=array([ 8.03612441, 10.39732102,  7.18124012, 14.19140621, 11.03775106,
        9.76523433]), pvalue=array([3.33064768e-15, 7.82725300e-24, 1.58422261e-12, 6.31491577e-41,
       1.84398096e-26, 2.35407544e-21]), df=array([798., 798., 798., 798., 798., 798.]))

**Challenge 2**

In this challenge, we will be working with california-housing data. The data can be found here:
- https://raw.githubusercontent.com/data-bootcamp-v4/data/main/california_housing.csv

In [87]:
df2 = pd.read_csv("https://raw.githubusercontent.com/data-bootcamp-v4/data/main/california_housing.csv")
df2.head()

longitude  latitude  housing_median_age  total_rooms  total_bedrooms  \
0    -114.31     34.19                15.0       5612.0          1283.0   
1    -114.47     34.40                19.0       7650.0          1901.0   
2    -114.56     33.69                17.0        720.0           174.0   
3    -114.57     33.64                14.0       1501.0           337.0   
4    -114.57     33.57                20.0       1454.0           326.0   

   population  households  median_income  median_house_value  
0      1015.0       472.0         1.4936             66900.0  
1      1129.0       463.0         1.8200             80100.0  
2       333.0       117.0         1.6509             85700.0  
3       515.0       226.0         3.1917             73400.0  
4       624.0       262.0         1.9250             65500.0

**We posit that houses close to either a school or a hospital are more expensive.**

- School coordinates (-118, 37)
- Hospital coordinates (-122, 34)

We consider a house (neighborhood) to be close to a school or hospital if the distance is lower than 0.50.

Hint:
- Write a function to calculate euclidean distance from each house (neighborhood) to the school and to the hospital.
- Divide your dataset into houses close and far from either a hospital or school.
- Choose the propper test and, with 5% significance, comment your findings.
 

In [100]:
# from math import sqrt

# def euclidean_distance_school(x1, y1, x2, y2):
#     """
#     Calculate the Euclidean distance between two points in a two-dimensional space.
    
#     Parameters:
#         x1 (float): x-coordinate of the first point.
#         y1 (float): y-coordinate of the first point.
#         x2 (float): x-coordinate of the second point.
#         y2 (float): y-coordinate of the second point.
        
#     Returns:
#         float: Euclidean distance between the two points.
#     """
#     return sqrt((x2 - x1)**2 + (y2 - y1)**2)

# # # Example usage:
# # x1, y1 = 1, 2  # Coordinates of the first point
# # x2, y2 = 4, 6  # Coordinates of the second point
# # distance = euclidean_distance_school(x1, y1, x2, y2)
# print("Euclidean distance:", distance)


# from math import sqrt

# def euclidean_distance_hospital(x1, y1, x2, y2):
#     """
#     Calculate the Euclidean distance between two points in a two-dimensional space.
    
#     Parameters:
#         x1 (float): x-coordinate of the first point.
#         y1 (float): y-coordinate of the first point.
#         x2 (float): x-coordinate of the second point.
#         y2 (float): y-coordinate of the second point.
        
#     Returns:
#         float: Euclidean distance between the two points.
#     """
#     return sqrt((x2 - x1)**2 + (y2 - y1)**2)

# # Example usage:
# x1, y1 = 1, 2  # Coordinates of the first point
# x2, y2 = 4, 6  # Coordinates of the second point
# distance = euclidean_distance_hospital(x1, y1, x2, y2)
# print("Euclidean distance:", distance)

Euclidean distance: 5.0
Euclidean distance: 5.0


In [104]:
# school_log =-118
# school_lat =37

# euclidean_distance_school((df2['longitude']),(df2['latitude']),school_log,school_lat )
# # hospital_distance =


TypeError: cannot convert the series to <class 'float'>